src/internal/components/pluggableVisualizations/geoChart/PluggableGeoPushpinChart.tsx

```diff
@@ -110,11 +112,12 @@ export class PluggableGeoPushpinChart extends PluggableBaseChart {
             ? secondaryMeasures
             : allMeasures.filter((measure: IBucketItem): boolean => !includes(sizeMeasures, measure))
         ).slice(0, this.getPreferedBucketItemLimit(BucketNames.COLOR));
+        this.locationItems = this.getLocationItems(buckets);
```

Comment:
IMO, we should not persist `locationItems ` to `this or PluggableGeoPushpinChart itself`.
Instead, it could be
```
private getSegmentItems(buckets: IBucket[], locationItems: IBucketItem[]): IBucketItem[] {...}
```

Reply:
By store locationItems as a property, the location items is calculated only one time and reuse later.
is there any reason to avoid use property?

Reply:
I'm afraid of we could miss updating `this.locationItems` at some points and it would cause future issue.
So passing it around is more stable solution.